.github/workflows/publish.yml

```diff
@@ -0,0 +1,81 @@
+name: publish
+
+on:
+  workflow_call:
+    inputs:
+      RELEASE_NAME:
+        type: string
+      ENVIRONMENT:
+        type: string
+      LATEST:
+        type: boolean
+      NIGHTLY:
+        type: boolean
+    secrets:
+      AWS_REGION:
+        required: true
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      RELEASES_BUCKET:
+        required: true
+      HELM_BUCKET:
+        required: true
+      GEMFURY_TOKEN:
+        required: true
+      DOCKERHUB_USERNAME:
+        required: true
+      DOCKERHUB_PASSWORD:
+        required: true
+      TELEMETRY_WRITE_KEY:
+        required: true
+
+jobs:
+  publish-artifacts:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.ENVIRONMENT }}
+    steps:
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: ${{ secrets.AWS_REGION }}
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      - uses: actions/download-artifact@v3
+        with:
+          name: binaries
+      - run: |
+          for PACKAGE in *.deb *.rpm; do
+            curl -fs -F package=@$PACKAGE https://${{ secrets.GEMFURY_TOKEN }}@push.fury.io/infrahq/
+          done
+        if: ${{ inputs.ENVIRONMENT == 'production' }}
+      - run: aws s3 sync . s3://${{ secrets.RELEASES_BUCKET }}/infra/v${{ inputs.RELEASE_NAME }} --exclude '*' --include '*.zip' --include '*.deb' --include '*.rpm' --include 'infra-checksums.txt' --acl public-read
+      - uses: actions/download-artifact@v3
+        with:
+          name: helm
+      - run: aws s3 sync . s3://${{ secrets.HELM_BUCKET }} --exclude '*' --include 'index.yaml' --include '*.tgz' --acl public-read
+      - run: |
+          for REPO in homebrew-tap scoop; do
```

Comment:
Deos this provide benefits to sticking with `goreleaser` for publishing `homebrew-tap` and `scoop` updates (not sure which options it has for publishing pre-built binaries)? Thinking from a maintainability standpoint on bash scripts + the multi-repo action events

Reply:
This allows more control over how artifacts are published. From what I can find, `goreleaser` doesn't support publishing without building. Coupling the build with publish means it's more difficult to migrate build tools or publishers and constrains  artifacts to whatever goreleaser supports. The release process already implements a custom publisher to archive deb and rpm to Gemfury which is essentially a bash script implemented in yaml. This only takes it a small step further.

The multi-repo action was unnecessary and has been replaced with a checkout of the target repo to invoke the update script. It's a good idea to keep the update script in the target repo since it's tightly coupled with the recipe/manifest.